packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -280,7 +280,7 @@ export abstract class RetrieveExecutor<T> extends DeployRetrieveExecutor<T> {
   protected async postOperation(
     result: RetrieveResult | undefined
   ): Promise<void> {
-    if (result) {
+    if (result?.response?.fileProperties !== undefined) {
```

Comment:
not sure if my comment got lost, so reposting.
not knowing the code well enough, a naive question. why is check fileProperties !== undefined not before calling setPropertiesForFilesRetrieve?

Also can setPropertiesForFilesRetrieve be called in other places? which means does this check need to go in setPropertiesForFilesRetrieve ?

Reply:
Also as per this fix, everything after line 283 will be skipped even if result is not null . is this expected?

Reply:
As of now, `setPropertiesForFilesRetrieve()` is not called anywhere else, according to a search.
![Screenshot 2024-02-21 at 12 33 16 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/139700604/b27f8fa2-d9d0-406c-91a7-fe50353cdd35)

Reply:
@diyer That is a good point. The check `if (result?.response?.fileProperties !== undefined) {` guarantees the block will not be executed if any property named is undefined. The previous block ran when result was not undefined, it might be useful to better understand which parts of result are used within the block, including the `createOutput` call

Reply:
There's a slight difference in the contents of the Output tab.  If the check is moved down to Line 289, there will be an extra blank line in the Output tab before the ended statement because output is an empty line and channelService appends it.

That's the only difference that I see - result is not used anywhere else in that block.

![Screenshot 2024-02-21 at 12 53 09 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/139700604/b4d5bbda-1278-47f3-80bc-6236c4c59105)
<img width="1307" alt="Screenshot 2024-02-21 at 12 52 37 PM" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/139700604/8739b092-fb33-4c3a-8a82-cf1f785390c3">